crypto/asn1/a_sign.c

```diff
@@ -131,12 +131,17 @@ int ASN1_item_sign_ex(const ASN1_ITEM *it, X509_ALGOR *algor1,
 {
     int rv = 0;
     EVP_MD_CTX *ctx = evp_md_ctx_new_ex(pkey, id, libctx, propq);
+    char mdname[80] = "";
+
+    if (EVP_PKEY_get_default_digest_name(pkey, mdname, sizeof(mdname)) > 0
+            && strcmp(mdname, "UNDEF") == 0) /* at least for Ed25519, Ed448 */
```

Comment:
Is this strcmp going to bite us? Should we use the wrapper?

Reply:
Also the code in `apps.c`, quoted above, and many other place in in `crypto/` use `strcmp()`.
Which wrapper do you mean? I cannot find one for `strcmp()`, and don't think any is needed.

Reply:
I think @romen means OPENSSL_strcasecmp which is unnecessary here as we don't do case insensitive comparison